### DISCUSSION TITLE
how to set ttl in keycloak admin console?

### DISCUSSION THREAD

**Discussion by garypan1990** [mesutpiskin/keycloak-2fa-email-authenticator#38] [General]
HI,
I would like to know how to set the TTL in the Keycloak admin console because I am using email OTP authentication. I received an email containing the code, but it does not show the TTL as depicted in the attached picture.

keycloak version: 21.1.0
![截圖 2024-05-09 中午12 37 41](https://github.com/mesutpiskin/keycloak-2fa-email-authenticator/assets/41365181/c783c038-c07e-4e8a-b801-cca67f9055d5)

Thank you.

**Comment by mesutpiskin** [comment]
You can set the TTL (and other parameters) directly from the Keycloak admin console under your authentication flow configuration.

Go to **Authentication → your flow → Email OTP execution → gear icon (⚙) → Config** and you will see fields like:

- **Time-to-Live (seconds)** — how long the code is valid (default: 300 s = 5 minutes)
- **Code Length** — number of digits (default: 6)
- **Resend Cooldown (seconds)** — wait time before a new code can be requested

Note: version 21.1.0 is very old and may not have all these config fields. The current release is v26.3.1, which supports Keycloak 26.5.x – 26.6.x. Please upgrade to a supported version for the full feature set.
